amespahdbpythonsuite/spectrum.py

```diff
@@ -362,3 +363,94 @@ def resample(self, grid: np.ndarray) -> None:
             self.data[uid] = s.flux.value
 
         self.grid = grid
+
+    def mcfit(self, obs, nsamples, **keywords) -> Mcfitted:
```

Comment:
Provide the same interface as fit, i.e., allows for both providing an Observation object or a two lists. Implement the option to also draw from a uniform distribution. 'nsamples' -> 'samples'.

Also annotate the method-parameters:

```def mcfit(self, y:  Union[Observation, list], yerr: list = list(), samples: int = 1024, uniform: bool = False, notice: bool = True, **keywords) -> MCFitted```